handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -796,4 +805,54 @@ protected static PKCS8EncodedKeySpec generateKeySpec(char[] password, byte[] key
 
         return encryptedPrivateKeyInfo.getKeySpec(cipher);
     }
+
+    /**
+     * Generates a new {@link KeyStore}.
+     *
+     * @param certChainFile a X.509 certificate chain file in PEM format,
+     * @param keyFile a PKCS#8 private key file in PEM format,
+     * @param keyPasswordChars the password of the {@code keyFile}.
+     *                    {@code null} if it's not password-protected.
+     * @return generated {@link KeyStore}.
+     */
+    protected static KeyStore buildKeyStore(File certChainFile, File keyFile, char[] keyPasswordChars)
```

Comment:
Probably better making it package-private? The subclasses are all in the same package already.